### PR TITLE
feat(moe): add DeepSeek dense fallback conformance and metadata

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

- validates Mobius' standard-op dense MoE fallback against a tiny NumPy DeepSeek-style reference
- fixes grouped routing so excluded experts are masked with `-inf` and uses the correct group reduction for `noaux_tc` versus group-limited greedy
- emits generic `model.mixture_of_experts` metadata for routed/shared expert counts, top-k, expert widths, graph representation, activation, and router semantics
- contains no model-name-gated metadata logic

The matching typed schema is on `justinchuby/onnx-genai:squad/hythe-deepseek-moe-phase1`.

## Validation

- 27 focused MoE/metadata tests passed
- 54 broader DeepSeek/MoE/auto-export tests passed (31 deselected)
- synthetic ONNX Runtime CPU conformance passed at `rtol=1e-5`, `atol=1e-6`
- Ruff check passed on changed files (using repository selectors; installed Ruff predates `RUF067`)

## Follow-up

Fused `com.microsoft::MoE`/`QMoE` emission, QMoE packing compatibility, and fused-vs-dense differential tests remain for the rest of Phase 1.

### CODE REVIEW

**Comment by justinchuby** [comment]
CI investigation update:

- Fixed Lint by applying the repository Ruff formatting to the changed DeepSeek metadata/test files.
- Added MoE metadata edge-case coverage (dense config, simple Top-K/defaults, explicit shared size, invalid expert count, missing intermediate size) and moved optional schema lookup after metadata construction so the new metadata setup remains covered when the sibling schema is unavailable. Local affected tests: 13 passed, 1 optional schema test skipped.
- Integration (fast), L4 Golden Comparison, and L5 Generation E2E are runner-infrastructure failures, not code regressions: all three fail during test collection with `ImportError: libcudart.so.13: cannot open shared object file`. Main had the identical Integration failure in runs 29940728850, 29927656222, and 29914787502 (for example job 88993606667). No test body from this PR executes in those jobs.

Pushed commits c69ae33 and 78d27b7 to the existing PR branch. Leaving the PR unmerged for Justin.

**Comment by justinchuby** [comment]
Follow-up main-branch comparison: the latest scheduled main L4 run 29892688925 (job 88836143900) and L5 run 29895209032 (job 88843797834) fail with the exact same collection-time `ImportError: libcudart.so.13` as this PR. This confirms Integration/L4/L5 are pre-existing GPU runner infrastructure issues. On the new PR run, Lint and codecov/patch are now passing; after the bounded 10-minute wait, Windows 3.11 plus Integration/L4/L5 were still pending. PR remains open and unmerged.

**Comment by justinchuby** [comment]
Status update from Sapper:

- Addressed all current actionable review threads: validate grouped expert divisibility/topk_group bounds and clamp noaux_tc group TopK to width 1; relax the brittle exact MatMul count; support `moe_num_shared_experts` / `num_shared_expert` aliases (including sequence values) in metadata.
- Added focused edge-case tests. Ruff check + format clean; 19 affected tests passed.
- Pushed `e27bf39` to `squad/hythe-deepseek-moe-phase1` without force.

CI was green aside from the documented runner-infrastructure L4/L5 failures; comments are addressed. Awaiting Justin's review approval + merge. I did not merge this PR.

**Comment by justinchuby** [comment]
Follow-up: GitHub reported the branch had become conflicting with current `main` after the review-fix push. I merged current `main`, resolved the sole `MoELayer` documentation conflict while retaining both the portable dense fallback contract and main's QMoE path, and pushed `62e98dc` without force.

Post-merge validation: Ruff check + format clean; 84 affected DeepSeek/MoE/metadata/GQA tests passed. Still not merged; awaiting Justin.

File: src/mobius/models/deepseek.py [hunk 1]
```diff
@@ -114,10 +114,14 @@ def _group_topk_selection(self, op, scores_for_choice):
...
         scores_grouped = op.Reshape(flat, [0, self.n_group, experts_per_group])
-        # Group score = sum of top-2 within each group
-        k_two = op.Constant(value_ints=[2])
-        group_top2, _ = op.TopK(scores_grouped, k_two, axis=-1, _outputs=2)
-        group_scores = op.ReduceSum(group_top2, [-1], keepdims=False)  # (B*S, n_group)
+        if self.topk_method == "noaux_tc":
+            # Bias-corrected routing scores groups by their two strongest experts.
+            k_two = op.Constant(value_ints=[2])
+            group_top2, _ = op.TopK(scores_grouped, k_two, axis=-1, _outputs=2)
+            group_scores = op.ReduceSum(group_top2, [-1], keepdims=False)
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/deepseek.py:156]: Fixed in e27bf39 and retained after the main merge in 62e98dc: grouped routing now validates expert divisibility/topk_group bounds and uses min(2, experts_per_group) for noaux_tc. Focused edge tests were added.

**Comment by justinchuby** [comment]
Patch-coverage follow-up: added graph coverage for one-expert noaux_tc groups plus both invalid topk_group bounds; 23 focused routing/metadata tests pass. Pushed `6194520` without force.

**Comment by justinchuby** [comment]
Latest Copilot follow-ups are addressed in `40846bb`: unset activation metadata defaults to SiLU, and grouped metadata now validates positive/divisible group counts and topk_group bounds. 28 focused tests pass; no unresolved current review threads remain.
